src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs

```diff
@@ -87,8 +83,8 @@ private SafeDirectoryHandle CreateDirectoryHandle(string path)
 
         private void CloseDirectoryHandle()
         {
-            _directoryHandle?.Dispose();
-            _directoryHandle = null;
+            Interop.Sys.CloseDir(_directoryHandle);
```

Comment:
You're not doing the Interlocked like on Windows because calling closedir twice is likely to not cause any problems because you're assuming the handle values are more widely distributed on Unix?

In general I think we don't use Interlocked for closing Windows handles.

Reply:
> assuming the handle values are more widely distributed on Unix?

In my experience they're usually not and file descriptors are much more likely to be recycled quickly. I can't speak to this particular readdir case.

Reply:
We're in a lock or constructor for all of the closes.

Reply:
Right, but you're still using Interlocked for windows. sorry to be pendantic I'm just trying to understand the reasoning.

Reply:
I'll interlock it